### DISCUSSION TITLE
What is direction of Asp.Net.Core?

### DISCUSSION THREAD

**Discussion by fairking** [dotnet/aspnetcore#28616] [Q&A]
**I am not interesting** in any of those render features. I would like to have a **simple, clear and tiny** Asp.Net.Core for providing an API for my **Vuejs application**. 
Currently I am using **3.1**. I'm glad the Asp.Net.Core became much **lighter and faster** than the previous Asp.Net with WebForms one. But recently I see things go in overcomplicated direction again adding Razor, Blazor, Azure, Auth, Annotations and other unnecessary features on top.

Will ASP.NET.Core v5 be as a simple tiny api server framework, or I should look for something else? Or will be there any core build one?

Is there any reason I should switch to v5? 
Is there any improvements in v5? Apart of Brazor as an anti-feature (imho)?

Please tell I am wrong and everything comes separately. 

Thanks.

**Comment by Tratcher** [comment]
See https://devblogs.microsoft.com/dotnet/announcing-net-5-0/ and https://devblogs.microsoft.com/aspnet/announcing-asp-net-core-in-net-5/ for a longer list of .NET 5 improvements. The short list is language improvements, performance improvements, and protocol improvements (HTTP/2, gRPC). Yes we spent a lot of effort on blazor, but if you don't need that then you don't have to use it.

One of ASP.NET Core's main tenants has always been pay-for-play. In other words, yes we have a lot of features, but they should only cost you something if you use them. If you want to use a minimal subset of the framework then go for it, it should be small and fast for you. We've even worked on things like trimming to make it smaller (https://devblogs.microsoft.com/dotnet/app-trimming-in-net-5/).

Here's some work we're tracking for 6.0 to make simple scenarios even simpler: https://github.com/dotnet/aspnetcore/issues/27724

**Comment by KieranDevvs** [comment]
Its not a simple answer to be honest. AspNetCore is pretty bare bones and any features you want to use can be added in at the cost of performance. The problem is that each extra component has its own complexity. Adding in MVC adds in quite a big overhead even if you just want a small segment of it (this is being improved). Whereas adding in something like SingalR adds very little overhead other than what you expect.

Best way to determine if its for you, is to decide what features you need and try them out. But as a general rule, it will be suitable for 99.9% of applications unless you're doing something ***extremely*** niche.

In your case, you probably want to check out: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/static-files?view=aspnetcore-5.0
